src/explore-education-statistics-admin/src/components/RouteLeavingGuard.tsx

```diff
@@ -0,0 +1,72 @@
+import ModalConfirm from '@common/components/ModalConfirm';
+import useToggle from '@common/hooks/useToggle';
+import React, { useEffect, useState } from 'react';
+import { Prompt, useLocation } from 'react-router';
+import { useHistory } from 'react-router-dom';
+
+interface Props {
+  blockRouteChange: boolean;
+}
+
+const RouteLeavingGuard = ({ blockRouteChange = false }: Props) => {
```

Comment:
Think we should be able to write some unit tests for this?

Reply:
Yep!  Not sure about the way I've changed route in the tests (putting a button in and clicking it), but I couldn't find another way that worked, tried using history.push but couldn't get it to work.

Reply:
Yep this approach is fine :smile: An alternative would have been to use `history.push` directly as the component still functions correctly. I've added this in as a test case and also cleaned up some of the test assertions to be a little more thorough.